Add support for handling translating keys with dots

jslang opened the pull request:
Additionally refactor pattern construction to allow key and ignored key pattern construction to share logic

Fixes: https://github.com/yola/gulp-static-i18n/issues/25

jslang committed:
Add support for handling translating keys with dots

Additionally refactor pattern construction to allow key and ignored key pattern construction to share logic

stefanor commented on lib/translators/json.js:
```diff
@@ -158,23 +158,33 @@ function getKeyFilter(keys, ignored) {
-  // using a bullet char to avoid conflict with keys with periods
-  reYes = reYes.replace(/\./g, '•');
+    // tokenize escaped dots to avoid confusing them with nested dot notation
+    re = re.replace(/\\\./g, '‣');
```
You could use a negative lookback assertion rather than a temporary unicode substitude.

This is just compounding existing hacks. @beck: Have you considered using a different data structure to strings, to represent the JSON path? e.g. an array of keys. I'm guessing it would significantly complicate the filtering though...

beck replied:
Workin on it now.

jslang replied:
I don't see a way to do lookback assertion in javascript? That was my first instinct. Is there a way to do it?

beck replied:
Yeah they're weird but can be done: https://github.com/yola/gulp-static-i18n/pull/27/files#diff-386ec0e3e845bca5691f1e4e2a49c817R135

beck committed:
Rename param `keys` to `lookups`

"keys" is a poor name because it implies an array of json keys. It is an
array of lookups, keys concatenated together using a specific lookup pattern.

beck committed:
Cleanup json key sibling test extraction

beck committed:
Refactor json's getKeyFilter

Refactor includes:
* new (hopefully better) variable names
* added documentation
* use of regex negative lookbehinds vs temporary tokenization

beck committed:
Update readme for escaping

beck committed:
Update changelog

beck committed:
Merge pull request #27 from yola/doug/refactor

A bit of refactoring and documentation

beck reviewed (approved):
Tested with a consumer. All looks good.

Bump and cut a new release.

Bump the minor version (not just the patch). Technically this is backward incompatible, but if someone is using backslashes in their json keys they have bigger problems than this.
